collector/container/scripts/download-support-package.sh

```diff
@@ -30,6 +30,8 @@ main() {
 
     zip_file="$(basename "${support_pkg}")"
 
+    cd "${TARGET_DIR}"
```

Comment:
WDYT if the `TARGET_DIR` **is** `$(workspaces.source.path)/source/staging` and you both create and enter it here?

```suggestion
    mkdir -p "${TARGET_DIR}"
    cd "${TARGET_DIR}"
```

Then, at the end of the script, you just rename the zip file.

There's no need to worry about interrupted partial downloads because (a) sha256sum check would not let partial file go through and besides (b) no-zero exit code from `curl` or anywhere of this script will abort the Tekton pipline with error.

I bring this up because it feels _a bit arbitrary_ that files are first downloaded to `$(workspaces.source.path)/source` to be later moved to `$(workspaces.source.path)/source/staging`.
From the perspective of this script's code it will be just cleaner if the caller provides it the ultimate directory path where everything happens and the script does not need to worry that at some point `/staging/` should be appended to the path.